revision to gene conversion API (#1355)
popsim-consortium/stdpopsim: docs/api.rst

.. autoclass:: stdpopsim.Population()
    :members:

> 

Author: I'm not sure this is the right place to put this?

    a contig with `use_species_gene_conversion=True` will result in a Contig
    with a larger "recombination rates" than otherwise, because these rates
    include gene conversion as well as crossing over.

Different engines implement gene conversion in different ways: at time
of writing, msprime only allows a *constant* rate of gene conversion,
while SLiM only allows gene conversion to be a fixed fraction of double-stranded
break events. So, in SLiM we use the recombination map for the local
rate of double-stranded breaks along the genome, while in msprime
we specify a constant rate of gene conversion so that the average total number
is the same as we would get from SLiM.

In principle, **bacterial recombination** is a completely different issue,
since bacterial recombination is by horizontal transfer of genomic segments.
However, bacterial recombination is at present implemented using the engine's
gene conversion mechanisms. So, bacterial species have the `bacterial_recombination`
flag set, and need `gene_conversion_length` to be defined. (It would be more accurate
to call this the "horizontal transfer segment length", but that would add another

> option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`

Reviewer: Unfortunately, there is no non-confusing term here to describe this. Some call it (bacterial) homologous recombination, and some refer to it as horizontal gene transfer even if it occurs within the same taxa/species. Maybe the best solution is to simply delete the "more accurate" and replace it with:
"It could also be called horizontal transfer or homologous recombination segment length, but ... "

> option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`

Author: I was worried about this also but
![Screenshot from 2022-09-13 17-06-20](https://user-images.githubusercontent.com/1046249/190030532-15375ab7-0f6f-4547-9075-8f114c3fdd38.png)

> option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`

Reviewer 2: 😐

> flag set, and need `gene_conversion_length` to be defined. (It would be more accurate
> to call this the "horizontal transfer segment length", but that would add another
> option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`

Author: ```suggestion
flag set, and need `gene_conversion_length` to be defined. (It could also be called horizontal
transfer or homologous recombination segment length, but that would add another
option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`
```

> option.) For such species, gene conversion does not happen, so `gene_conversion_fraction`

Author: whoops - for some reason I didn't see Franz's comment before I posted this!
